pkg/route/registry/route/rest.go

```diff
@@ -13,16 +13,19 @@ import (
 
 	"github.com/openshift/origin/pkg/route/api"
 	"github.com/openshift/origin/pkg/route/api/validation"
+	rac "github.com/openshift/origin/pkg/route/controller/allocation"
 )
 
 // REST is an implementation of RESTStorage for the api server.
 type REST struct {
-	registry Registry
+	registry  Registry
+	allocator rac.RouteAllocationController
```

Comment:
Generally you wouldn't pass a controller in - just an interface.  You can define an interface here that matches the controller plugin